eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -218,13 +225,48 @@ foreach ($service in $serviceNameList) {
 }
 
 # Core packages belong under the "Other" node in the ToC
-$otherPackageItems = @()
+$otherPackageItems = New-Object -TypeName System.Collections.Generic.List[PSCustomObject]
 $otherPackages = $packagesForToc.Values.Where({ $_.ServiceName -eq 'Other' })
-$otherPackages = $otherPackages | Sort-Object -Property Package
+$otherPackages = $otherPackages | Sort-Object -Property DisplayName
 
 if ($otherPackages) {
   foreach ($otherPackage in $otherPackages) {
-    $otherPackageItems += GetClientPackageNode $otherPackage
+    $segments = $otherPackage.DisplayName.Split(' - ')
+
+    if ($segments.Count -gt 1) {
+      $currentNode = $otherPackageItems
+      foreach ($segment in $segments[0..($segments.Count - 2)]) {
```

Comment:
Why count - 2?

Reply:
I know you added the comment to help address this question, but I think I'm still missing something. Maybe give an example of a display name that would illustrates what you are trying to do here. 

If I understand correct we might end up with something like "Core - Http" and is the goal that we essentially drop the Http part and because we want that to be a leaf node and not nested node?

Reply:
Correct. This loop does not build the leaf nodes. It only builds the branches extending from the "Other" node up to where the leaf should be added. 

In the case of DisplayName "Core - Http" 

```yaml
# $otherPackages 
- name: Other
  landingPageType: Service
  items:

   # Built by the loop 
    - name: Core 
      landingPageType: Service
      items: 

        # Added outside of the loop
        - name: Http
          href: ~/some/md/file.md
          children: 
            - @azure/core-http
 ```

In the case of DisplayName "Core - Client - Http" 

```yaml
# $otherPackages 
- name: Other
  landingPageType: Service
  items:

   # Built by the loop 
    - name: Core 
      landingPageType: Service
      items: 
        # Also built by the loop 
        - name: Client 
          landingPageType: Service
          items: 

            # Added outside of the loop
            - name: Http
               href: ~/some/md/file.md
               children: 
                 - @azure/core-http
 ```